server/datastore/mysql/software_installers.go

```diff
@@ -166,6 +173,14 @@ INSERT INTO software_installers (
 
 	id, _ := res.LastInsertId()
 
+	// TODO: how does should this check work in the context of editng an existing software installer to
+	// remove existing labels (i.e. switching from custom targets to all hosts)?
```

Comment:
If the validations passed, I think we should always end up calling `upsertSoftwareInstallerLabels`? And if there are none in the new payload, it removes any existing ones.